### PR TITLE
[CELEBORN-521] correct exception and unify unRetryableException

### PR DESCRIPTION
**Pull request by RexXiong**
…rtitionUnRetryableException

### What changes were proposed in this pull request?
Correct exception and unify unRetryableException

### Why are the changes needed?
Currently Some Exception Type is not correct for client to identify what should do next. Such as throw RuntimeException when file corrupted in MapDataPartitionReader / OpenStream when FileNotFoundException. We can unify to PartitionUnRetryableException for reader can give up this partition and choose to regenerate the partition data, And this would benefit Flink to regenerate upstream data.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Tpcds & Random worker kill.

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala [hunk 1]
```diff
@@ -36,6 +36,7 @@ import org.apache.celeborn.common.network.protocol.Message.Type
 import org.apache.celeborn.common.network.server.{BaseMessageHandler, ChunkStreamManager, CreditStreamManager}
 import org.apache.celeborn.common.network.util.{NettyUtils, TransportConf}
 import org.apache.celeborn.common.protocol.PartitionType
+import org.apache.celeborn.common.util.ExceptionUtils
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala:39]: import org.apache.celeborn.common.exception.CelebornException
can be deleted in Line29

**Comment by waitinfuture** [comment]
Seems this PR only wraps IOException to PartitionUnRetryAbleException, then why not just use IOException instead of PartitionUnRetryAbleException?

**Comment by RexXiong** [comment]
> Seems this PR only wraps IOException to PartitionUnRetryAbleException, then why not just use IOException instead of PartitionUnRetryAbleException?

Client may encounter other IOException like Network error, PartitionUnRetryAbleException means the data can not used anymore by server side knowledge.

**Comment by RexXiong** [comment]
How about current? @waitinfuture @FMX

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Comment by waitinfuture** [comment]
I wonder if check for Exception type is a good style.

**Pull request merged**